build/src/voices.js

```diff
@@ -0,0 +1,357 @@
+export function filterOnQuality(voices, quality) {
+    quality = Array.isArray(quality) ? quality : [quality];
+    return voices.filter(({ quality: voiceQuality }) => {
+        return quality.some((qual) => qual === voiceQuality);
+    });
+}
+export function filterOnNovelty(voices) {
+    return voices.filter(({ name }) => {
+        return !novelty.includes(name);
+    });
+}
+export function filterOnVeryLowQuality(voices) {
+    return voices.filter(({ name }) => {
+        return !veryLowQuality.find((v) => name.startsWith(v));
+    });
+}
+function updateVoiceInfo(recommendedVoice, voice) {
+    voice.label = recommendedVoice.label;
+    voice.gender = recommendedVoice.gender;
+    voice.recommendedPitch = recommendedVoice.recommendedPitch;
+    voice.recommendedRate = recommendedVoice.recommendedRate;
+    return voice;
+}
+export function filterOnRecommended(voices, _recommended = recommended) {
+    const voicesRecommended = [];
+    const voicesLowerQuality = [];
+    recommendedVoiceLoop: for (const recommendedVoice of _recommended) {
+        if (Array.isArray(recommendedVoice.quality) && recommendedVoice.quality.length > 1) {
+            const voicesFound = voices.filter(({ name }) => name.startsWith(recommendedVoice.name));
+            if (voicesFound.length) {
+                for (const qualityTested of ["high", "normal"]) {
+                    for (let i = 0; i < voicesFound.length; i++) {
+                        const voice = voicesFound[i];
+                        const rxp = /^.*\((.*)\)$/;
+                        if (rxp.test(voice.name)) {
+                            const res = rxp.exec(voice.name);
+                            const maybeQualityString = res ? res[1] || "" : "";
+                            const qualityDataArray = qualityTested === "high" ? highQuality : normalQuality;
+                            if (recommendedVoice.quality.includes(qualityTested) && qualityDataArray.includes(maybeQualityString)) {
+                                voice.quality = qualityTested;
+                                voicesRecommended.push(updateVoiceInfo(recommendedVoice, voice));
+                                voicesFound.splice(i, 1);
+                                voicesLowerQuality.push(...(voicesFound.map((v) => {
+                                    v.quality = "low"; // Todo need to be more precise for 'normal' quality voices
+                                    return updateVoiceInfo(recommendedVoice, v);
+                                })));
+                                continue recommendedVoiceLoop;
+                            }
+                        }
+                    }
+                }
+                const voice = voicesFound[0];
+                for (let i = 1; i < voicesFound.length; i++) {
+                    voicesLowerQuality.push(voicesFound[i]);
+                }
+                voice.quality = voicesFound.length > 3 ? "veryHigh" : voicesFound.length > 2 ? "high" : "normal";
```

Comment:
If we found three different variants, then quality should be set to `high` rather than `veryHigh`. With two variants the quality should be set to `normal`. With a single variant, it's impossible to know the quality and we should set the `quality` accordingly by using `undefined`.

Reply:
If I understand correctly, with three different variants, quality is set to `high` and not `veryHigh`. 

For less than one quality variant : quality is set to the first quality value in the array with a default to `undefined`

Reply:
Technically, we should implement this based on the number of values in quality.

With three values in quality, finding two voices means that one of the two voices is at least the second best option. Finding three voices means that we should use the best option available.

With two values in quality, finding two values means that we can set it to the highest quality listed.